pulsar/producer_partition.go

```diff
@@ -145,6 +145,7 @@ func newPartitionProducer(client *client, topic string, options *ProducerOptions
 	}
 	err := p.grabCnx()
 	if err != nil {
+		p.batchFlushTicker.Stop()
```

Comment:
The `grabCnx()` is called in two places in total, except here in the`reconnectToBroker()` logic, it looks like we need the same fix.

Reply:
but `newTicker()` is called only with `newPartitionProducer()`, i don't think we need to fix it.